### PR TITLE
DCOS-10878: Update CLI Install Instructions

### PR DESCRIPTION
**Pull request by MatApple**
Splice `dcos auth login` command step into the CLI Install instructions when authentication is enabled.

https://mesosphere.atlassian.net/browse/DCOS-10878

### CODE REVIEW

File: plugins/oauth/hooks.js [hunk 1]
```diff
@@ -62,6 +64,21 @@ module.exports = Object.assign({}, StoreMixin, {
...
+    let steps = instructions.split('&&');
+    steps.splice(-1, 0, ' \n dcos auth login ');
+
+    return steps.join('&&');
+  },
+
+  dcosInstallSteps(steps) {
+    let newSteps = steps.map((step) => step);
+
+    newSteps.splice(-1, 0, 'dcos auth login');
```
Comment (mlunoe) [inline plugins/oauth/hooks.js]: Could we get a comment here? something like: "Add login as the step second to last, before printing command options"

File: plugins/oauth/hooks.js [hunk 2]
```diff
@@ -62,6 +64,21 @@ module.exports = Object.assign({}, StoreMixin, {
     }]);
   },
 
+  dcosInstallCommand(instructions) {
+    let steps = instructions.split('&&');
+    steps.splice(-1, 0, ' \n dcos auth login ');
```
Comment (mlunoe) [inline plugins/oauth/hooks.js]: Could we get a comment here? something like: "Add login as the step second to last, before printing command options"

Comment (orlandohohmeier) [inline plugins/oauth/hooks.js]: This seems highly error prone to me and I suggest to refactor this using the `mount` package.

Comment (MatApple) [inline plugins/oauth/hooks.js]: Made some changes so we don't split a string and pass in an Array. I'm not sure a Mount would be more beneficial without code duplication. Do you have a particular example in mind @orlandohohmeier?

Comment (orlandohohmeier) [inline plugins/oauth/hooks.js]: I'd much rather duplicate code/data than having a hard to maintain and error prone implementation. DRY isn't absolute.

**Comment by mlunoe** [comment]
I would still like some comments

**Comment by mlunoe** [comment]
Run unit tests

**Comment by mlunoe** [comment]
Feel free to :shipit: once tests pass

**Comment by rcorral** [comment]
run unit tests

**Comment by rcorral** [comment]
run integration tests

**Comment by rcorral** [comment]
legit unit test errors.

**Comment by nLight** [comment]
Fixed the linting errors, merging it.

**Pull request merged**